--- a/lib/libipsecconf/Makefile
+++ b/lib/libipsecconf/Makefile
@@ -40,7 +40,9 @@ $(OBJS): parser.tab.h
 
 # Use UNIX basename to strip off the directory. Use grep to workaround flex < 2.5.35
 lex.yy.c: parser.l
-	cd $(builddir) && $(LEX) -t $(srcdir)/$$(basename $<) | grep -v '^extern int isatty' > $@
+	cd $(builddir) && $(LEX) -t $(srcdir)/$$(basename $<) | grep -v '^extern int isatty' | \
+	sed -e 's/for ( i = 0; i < _yybytes_len; ++i )$$/for ( i = 0; (int)i < _yybytes_len; ++i )/' \
+	-e "s#(yy_n_chars), (size_t) num_to_read )#(yy_n_chars), (int) num_to_read )#" > $@
 
 # Use wild card rule so that GNU Make knows that both are output from
 # a single recipe.
